ouroboros-network/src/Ouroboros/Network/PeerSelection/PeerSelectionActions.hs

```diff
@@ -97,13 +96,13 @@ withPeerSelectionActions
      , Ord peeraddr
      , Exception exception
      )
-  => StrictTVar m (Config peeraddr)
+  => StrictTVar m (Config PeerTrustable peeraddr)
   -> PeerActionsDNS peeraddr resolver exception m
-  -> PeerSelectionActionsArgs peeraddr peerconn exception m
-  -> WithLedgerPeersArgs m
+  -> PeerSelectionActionsArgs (CardanoPeerSelectionActions m) (CardanoLedgerPeersConsensusInterface m) (CardanoPublicRootPeers peeraddr) PeerTrustable peeraddr peerconn exception m
+  -> WithLedgerPeersArgs (CardanoLedgerPeersConsensusInterface m) m
   -> PeerSelectionActionsDiffusionMode peeraddr peerconn m
   -> (   (Async m Void, Async m Void)
-      -> PeerSelectionActions peeraddr peerconn m
+      -> PeerSelectionActions (CardanoPeerSelectionActions m) (CardanoPublicRootPeers peeraddr) PeerTrustable (CardanoLedgerPeersConsensusInterface m) peeraddr peerconn m
       -> m a)
```

Comment:
With some type aliases this could look simpler:
```hs
  -> CardanoPeerSelectionActionsArgs ...
  -> WithLedgerPeersArgs (CardanoLedgerPeersConsensusInterface m) m
  -> PeerSelectionActionsDiffusionMode peeraddr peerconn m
  -> (   (Async m Void, Async m Void)
      -> CardanoPeerSelectionActions ...
      -> m a)
```

ref: cb0da21a3ea31eee99c7348347bef519810b0258

Reply:
This signature ends up being fully polymorphic so I won't be adding this type alias here